### ISSUE TITLE
Finite vs infinite numbers

### ISSUE THREAD

**Issue by lassik** [pre-srfi/make-float#1]
The intention is that the procedures in this SRFI make finite numbers only; i.e. the result should satisfy `finite?` in all cases; i.e. it should not be infinite or NaN.

Are there some innocuous-looking arguments for which an infinity or Nan could result due to the finite precision of floating point? In those cases, should we:

* return a non-finite real
* return `#f`
* raise an exception

Likewise, when asked to break an infinity or a NaN into its component parts, what should we return?

**Comment by johnwcowan** [comment]
Well, yes, 1e400 is +0.inf and similarly 1e-400 is -0.inf.  That is taken
care of for you by the rules of float arithmetic.  For 400 read 40 for
single-floats.

Which leads to whether to support S, F, D, L as alternatives to E.  F is
normally single-float and D is normally double-float, as in Fortran, but
the meanings of S and L vary.  CL and Fortran default to single-float if
you use E, most other languages to double-float.

The only rules are that if a number is representable by a float type, it is
also representable by all float types to the right, and that if two types
have the same implementation, so do any intervening types.

**Comment by lassik** [comment]
Good point. We should definitely support S, F, D, L. Probably by taking an extra argument instead of having a different constructor procedure for each of them?

**Comment by johnwcowan** [comment]
Yes

**Comment by lassik** [comment]
Sorry, just moved the S F D L discussion to a new thread.
